solutionbox/inception/datalab_solutions/inception/_predictor.py

```diff
@@ -40,42 +42,40 @@ def _tf_predict(model_dir, image_files):
         image_bytes = ff.read()
         feed_dict[inputs['image_bytes']].append(image_bytes)
         feed_dict[inputs['key']].append(str(ii))
-    predictions, scores = sess.run([outputs['prediction'], outputs['scores']],
-                                   feed_dict=feed_dict)
-  return predictions, scores
+    predictions, labels, scores = sess.run(
+        [outputs['prediction'], outputs['labels'], outputs['scores']], feed_dict=feed_dict)
+  return predictions, labels[0], scores
 
 
-def predict(model_dir, image_files, labels_file):
+def predict(model_dir, image_files):
   """Local prediction."""
 
-  predictions, scores = _tf_predict(model_dir, image_files)
-  labels = _util.get_labels(labels_file)
-  labels.append('UNKNOWN')
-  labels_and_scores = [(labels[predicted_index], class_scores[predicted_index])
-                       for predicted_index, class_scores in zip(predictions, scores)]
-  return labels_and_scores
+  predictions, labels, scores = _tf_predict(model_dir, image_files)
+  labels = list(labels)
+  results = [(predicted, label_scores[labels.index(predicted)])
+             for predicted, label_scores in zip(predictions, scores)]
+  return results
 
 
-def batch_predict(model_dir, input_csv, labels_file, output_file, output_bq_table):
+def batch_predict(model_dir, input_csv, output_file, output_bq_table):
   """Local batch prediction."""
 
   input_csv_f = ml.util._file.read_file_stream(input_csv)
   reader = csv.reader(input_csv_f)
   image_files = [x[0] for x in reader]
-  predictions, scores = _tf_predict(model_dir, image_files)
-
-  labels = _util.get_labels(labels_file)
-  labels.append('UNKNOWN')
+  predictions, labels, scores = _tf_predict(model_dir, image_files)
```

Comment:
This is the work that I planned to do but forgot to add a TODO. Now it is batched!